src/context.ts

```diff
@@ -146,9 +148,8 @@ export class Context implements ContextValue {
     throw new SdkError("cache must be defined in Context for this feature to work.");
   }
 
-  get subgraph(): SubgraphConfiguration {
-    if (this.ctx.subgraph) return this.ctx.subgraph;
-    throw new SdkError("subgraph configuration must be defined in Context for this feature to work.");
+  get subgraph(): SubgraphConfiguration | undefined {
```

Comment:
As discussed in TG, I'd think that `null` is a better value to return
```suggestion
  get subgraph(): SubgraphConfiguration | null {
```